src/js/providers/tracks-mixin.js

```diff
@@ -428,6 +417,34 @@ define(['../utils/underscore',
         return trackId;
     }
 
+    function addVTTCuesToTrack(track, vttCues) {
+        if (this._renderNatively) {
+            var textTrack = this._tracksById[track._id];
+            // the track may not be on the video tag yet
+            if (!textTrack) {
+
+                if (!this._cuesByTrackId) {
+                    this._cuesByTrackId = {};
+                }
+                this._cuesByTrackId[track._id] = { cues: vttCues, loaded: false};
+                return;
+            }
+            // Cues already added
+            if (this._cuesByTrackId[track._id] && this._cuesByTrackId[track._id].loaded) {
+                return;
+            }
+
+            var cue;
+            this._cuesByTrackId[track._id] = { cues: vttCues, loaded: true };
+
+            while((cue = vttCues.shift())) {
+                _addCueToTrack(textTrack, cue);
+            }
+        } else {
+            track.data = vttCues;
+        }
```

Comment:
So if we don't render natively, we just set `track.data` to `vttcue`? If so this might be cleaner

```
if (!this._renderNatively) {
    track.data = vttCues;
    return;
}

var textTrack = this._tracksById[track._id];
.....

```